src/leanback/java/com/archos/mediacenter/video/leanback/details/VideoDetailsFragment.java

```diff
@@ -349,8 +350,16 @@ public void onItemClicked(Presenter.ViewHolder itemViewHolder, Object item, RowP
                 if(item instanceof ScraperTrailer){
                     // Breaks AndroidTV acceptance but needed to launch scraper in Youtube app instead of browser
                     Intent browserIntent = new Intent(Intent.ACTION_VIEW, ((ScraperTrailer)item).getUrl());
-                    startActivity(browserIntent);
-                    //WebUtils.openWebLink(getActivity(), ((ScraperTrailer)item).getUrl().toString());
+                    ActivityInfo activityInfo = browserIntent.resolveActivityInfo(getActivity().getPackageManager(), browserIntent.getFlags());
+                    
+                    if (activityInfo != null && !activityInfo.processName.equals("com.google.android.tv.frameworkpackagestubs")) {
+                        startActivity(browserIntent);
+                    }
+                    else {
+                        String url = ((ScraperTrailer)item).getUrl().toString().replace("https://www.youtube.com/watch", "https://www.youtube.com/tv#/watch");
+
+                        WebUtils.openWebLink(getActivity(), url);
```

Comment:
I had users complaining not using Android Youtube app to watch trailers hence 5d46e6c15b24bc617aa694aa1fca3ffb77614f68
But I like your tv#/watch trick which is enough to convince me.